webapp/packages/core-ui/src/Tabs/useTabDragAndDrop.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { useRef, useState } from 'react';
+import { useDrag, useDrop } from '@dbeaver/react-dnd';
+
+export interface ITabDragAndDropOptions {
+  tabId: string;
+  stateKey: string | undefined;
+  onReorder: ((draggedTabId: string, targetTabId: string, position: 'before' | 'after') => void) | null;
```

Comment:
we duplicate position: 'before' | 'after', probably we want to move it to separate type like TabReorderPosition